lib/public/views/Logs/Logs.js

```diff
@@ -27,37 +27,65 @@ export default class Overview extends Observable {
         this.filterCriteria = [];
         this.data = [];
         this.filtered = [];
+        this.error = false;
         this.headers = ['ID', 'Author ID', 'Title', 'Creation Time'];
-
-        this.fetchData();
     }
 
     /**
-     * Get the headers from the Overview class
-     * @returns {Array} Returns the headers
+     * Retrieve every relevant log from the API
+     * @returns {undefined} Injects the data object with the response data
      */
-    getHeaders() {
-        return this.headers;
+    async fetchAllLogs() {
+        const response = await fetchClient('/api/logs', { method: 'GET' });
+        const result = await response.json();
+
+        if (result.data) {
+            this.data = result.data;
+            this.filtered = [...result.data];
+            this.error = false;
+        } else {
+            this.error = true;
+        }
+
+        this.notify();
     }
 
     /**
-     * Fetch all relevant logs data from api
+     * Retrieve a specified log from the API
+     * @param {Number} id The ID of the log to be found
      * @returns {undefined} Injects the data object with the response data
      */
-    async fetchData() {
-        const response = await fetchClient('/api/logs', { method: 'GET' });
-        const result = await response.json();
-        this.data = result.data;
-        this.filtered = [...result.data];
-        this.notify();
+    async fetchLog(id) {
+        // Do not call this endpoint again if we already have all the logs
+        if (this.data.length < 1) {
```

Comment:
The `data` property can be undefined, an additional `if (this.data && ..` should make it a bit safer.